### PR TITLE
Reusable workflow for JAX-Pallas unit tests

### PR DESCRIPTION
**Pull request by DwarKapex**
The Jax, Pallas and Levanter unit-test workflows are almost identical, the difference is the way they run the unit-tests and process the test result data. To avoid code repeating and reduce the workflow count, it is reasonable to have only one unit-test workflow that wraps changeable part (test run script and test result processing) and outcomes the results in a standard way.

### CODE REVIEW

File: .github/workflows/_ci.yaml [hunk 1]
```diff
@@ -240,6 +255,21 @@ jobs:
     with:
       IMAGE: ${{ needs.build-pallas.outputs.DOCKER_TAG_FINAL }}
       TEST_NAME: pallas
+      EXECUTE: |
+        docker run --gpus all ${{ inputs.JAX_IMAGE }} test-jax.sh -b backend-independent | tee test-backend-independent.log;
```
Comment (yhtang) [inline .github/workflows/_ci.yaml:259]: jax->pallas?

File: .github/workflows/_ci.yaml [hunk 2]
```diff
@@ -250,7 +280,20 @@ jobs:
     uses: ./.github/workflows/_test_unit.yaml
     with:
       IMAGE: ${{ needs.build-levanter.outputs.DOCKER_TAG_FINAL }}
-      TEST_NAME: levanter
+      TEST_NAME: pallas
```
Comment (yhtang) [inline .github/workflows/_ci.yaml:283]: pallas->levanter?

File: .github/workflows/_ci.yaml [hunk 3]
```diff
@@ -250,7 +280,20 @@ jobs:
...
+      EXECUTE: |
+        docker run --gpus all --shm-size=1g ${{ inputs.LEVANTER_IMAGE }} bash -ec "pip install flake8 pytest test_util && pytest /opt/levanter/tests" | tee test-levanter.log
```
Comment (yhtang) [inline .github/workflows/_ci.yaml:285]: test_util->test-utils

**Review by yhtang — CHANGES_REQUESTED** [review]

File: .github/workflows/_ci.yaml [hunk 4]
```diff
@@ -205,6 +205,21 @@ jobs:
...
+        errors=$(cat test-*.log | grep -c 'ERROR:' || true);
+        failed_tests=$(cat test-*.log | grep -c 'FAILED in' || true);
+        passed_tests=$(cat test-*.log | grep -c 'PASSED in' || true);
+        total_tests=$((failed_tests + passed_tests));
+        echo "TOTAL_TESTS=${total_tests}" >> $GITHUB_OUTPUT;
+        echo "ERRORS=${errors}" >> $GITHUB_OUTPUT;
+        echo "PASSED_TESTS=${passed_tests}" >> $GITHUB_OUTPUT;
+        echo "FAILED_TESTS=${failed_tests}" >> $GITHUB_OUTPUT;
+      ARTIFACTS: |
+        test-backend-independent.log
```
Comment (yhtang) [inline .github/workflows/_ci.yaml:221]: Are sitrep/badge jsons uploaded as artifacts?

File: .github/workflows/_ci.yaml [hunk 5]
```diff
@@ -251,6 +280,19 @@ jobs:
     with:
       IMAGE: ${{ needs.build-levanter.outputs.DOCKER_TAG_FINAL }}
       TEST_NAME: levanter
+      EXECUTE: |
+        docker run --gpus all --shm-size=1g ${{ needs.build-levanter.outputs.DOCKER_TAG_FINAL }} bash -ec "pip install flake8 pytest test-util && pytest /opt/levanter/tests" | tee test-levanter.log
+      STATISTICS_SCRIPT: |
+        errors=$(cat test-*.log | grep -c 'ERROR:' || true)
+        failed_tests=$(cat test-*.log | grep -c 'FAILED in' || true)
```
Comment (yhtang) [inline .github/workflows/_ci.yaml:287]: This likely still suffers from the failure in:
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/7981652065/job/21794674559
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/7981652065/job/21794674761

**Review by yhtang — CHANGES_REQUESTED** [review]

File: .github/workflows/_ci.yaml (hunk 4 above)
Comment (DwarKapex) [reply .github/workflows/_ci.yaml:221]: Yes, sitrep/badge is a common files to upload, so I moved them to `_test_unit.yaml`.

File: .github/workflows/_ci.yaml (hunk 5 above)
Comment (DwarKapex) [reply .github/workflows/_ci.yaml:287]: Fixed

**Comment by yhtang** [comment]
Wait a sec. It seems that `test_utils` in the Levanter unit test actually refers to this: https://github.com/stanford-crfm/levanter/blob/main/tests/test_utils.py It could not be found previously because the pytest command was ran outside the `tests` folder. It's not that `test-utils` package from PyPI, which is decade-only and not maintained.

Could you please fix it?

**Comment by DwarKapex** [comment]
> Wait a sec. It seems that `test_utils` in the Levanter unit test actually refers to this: https://github.com/stanford-crfm/levanter/blob/main/tests/test_utils.py It could not be found previously because the pytest command was ran outside the `tests` folder. It's not that `test-utils` package from PyPI, which is decade-only and not maintained.
> 
> Could you please fix it?

Checking that out.

**Comment by yhtang** [comment]
Next time please wait for the presubmit to complete before merging...

**Pull request merged**